### PR TITLE
Fix issue when cell is empty

### PR DESCRIPTION
**Pull request by luke-biel**

### CODE REVIEW

File: src/ion/section.rs [hunk 1]
```diff
@@ -230,4 +230,17 @@ mod tests {
         let rows = ion.get("FOO").unwrap().rows_without_header();
         assert_eq!(1, rows.len());
     }
+
```
Comment (platatita) [inline src/ion/section.rs:287]: please add tests when there is a header and use `mod` here to divide tests per `with_` and `without_`

File: src/ion/section.rs [hunk 2]
```diff
@@ -230,4 +230,32 @@ mod tests {
...
+    #[test]
+    fn reads_properly_empty_first_column() {
```
Comment (platatita) [inline src/ion/section.rs]: could you add the case as well:
```
            [FOO]
             |    | -- | -y |
             | -1 | 2  | 3  |
             |    | y  | z  |
```

**Comment by luke-biel** [comment]
This branch benches: 
```
test parse::section_on_end_of_ion                      ... bench:   4,178,517 ns/iter (+/- 139,988)
test parse::section_on_end_of_ion_parser_no_prealloc   ... bench:   4,950,057 ns/iter (+/- 223,405)
test parse::section_on_end_of_ion_tuned_parser         ... bench:   3,600,501 ns/iter (+/- 208,444)
test parse::section_on_start_of_ion                    ... bench:   4,109,221 ns/iter (+/- 209,063)
test parse::section_on_start_of_ion_parser_no_prealloc ... bench:   4,923,804 ns/iter (+/- 350,779)
test parse::section_on_start_of_ion_tuned_parser       ... bench:   3,573,462 ns/iter (+/- 832,124)
test parse_filtered::section_on_end_of_ion             ... bench:     604,675 ns/iter (+/- 22,657)
test parse_filtered::section_on_start_of_ion           ... bench:      14,664 ns/iter (+/- 912)
```
vs master:
```
test parse::section_on_end_of_ion                      ... bench:   4,352,258 ns/iter (+/- 2,052,358)
test parse::section_on_end_of_ion_parser_no_prealloc   ... bench:   5,062,868 ns/iter (+/- 413,860)
test parse::section_on_end_of_ion_tuned_parser         ... bench:   3,894,922 ns/iter (+/- 1,387,257)
test parse::section_on_start_of_ion                    ... bench:   4,283,458 ns/iter (+/- 1,887,530)
test parse::section_on_start_of_ion_parser_no_prealloc ... bench:   5,060,836 ns/iter (+/- 1,876,817)
test parse::section_on_start_of_ion_tuned_parser       ... bench:   3,736,928 ns/iter (+/- 329,278)
test parse_filtered::section_on_end_of_ion             ... bench:     641,073 ns/iter (+/- 27,164)
test parse_filtered::section_on_start_of_ion           ... bench:      15,363 ns/iter (+/- 3,930)
```

File: src/ion/section.rs [hunk 3]
```diff
@@ -83,151 +84,274 @@ impl IntoIterator for Section {
...
     }
 
-    #[test]
-    fn into_iter_section() {
-        let mut ion = ion!(r#"
+    mod into_iter {
+        use super::*;
+
+        #[test]
+        fn ref_section() {
```
Comment (platatita) [inline src/ion/section.rs]: it should be called `get`

File: src/ion/section.rs [hunk 4]
```diff
@@ -83,151 +84,274 @@ impl IntoIterator for Section {
...
+            let section: &Section = ion.get("FOO").unwrap();
+            let rows: Vec<_> = section.into_iter().collect();
+            assert_eq!(3, rows.len());
+        }
 
-    #[test]
-    fn into_iter_section_loop() {
-        let mut ion = ion!(r#"
+        #[test]
+        fn section() {
```
Comment (platatita) [inline src/ion/section.rs]: `remove`

File: src/ion/section.rs [hunk 5]
```diff
@@ -83,151 +84,274 @@ impl IntoIterator for Section {
...
+            assert_eq!(3, rows.len());
         }
-        assert_eq!(3, rows.len());
-    }
 
-    #[test]
-    fn into_iter_section_with_header() {
-        let mut ion = ion!(r#"
+        #[test]
+        fn section_loop() {
```
Comment (platatita) [inline src/ion/section.rs]: `remove_and_iter`

File: src/ion/section.rs [hunk 6]
```diff
@@ -83,151 +84,274 @@ impl IntoIterator for Section {
...
-        assert!(rows.len() == 3);
+        disallowed_cell_contents.is_match(s)
     }
 
-    #[test]
-    fn into_iter_section() {
-        let mut ion = ion!(r#"
+    mod into_iter {
+        use super::*;
+
```
Comment (platatita) [inline src/ion/section.rs:134]: please divide it here `with_header` and `without_`

File: src/ion/section.rs [hunk 7]
```diff
@@ -83,151 +84,274 @@ impl IntoIterator for Section {
...
+                rows.push(row);
+            }
+            assert_eq!(3, rows.len());
+        }
 
-    #[test]
-    fn row_with_header() {
-        let ion = ion!(r#"
+        #[test]
+        fn section_with_header() {
```
Comment (platatita) [inline src/ion/section.rs]: it should be called `remove` within the mod `with_header`

**Review by Dzejkop — APPROVED** [review]

**Comment by luke-biel** [comment]
@platatita The `into_iter` was split into two submodules

**Pull request merged**